### PR TITLE
Explain why we do not one-hot encode in episode 4

### LINKED ISSUE #292: why not using one-hot encoding and softmax for CIFAR-10 model
In the [material](https://carpentries-incubator.github.io/deep-learning-intro/04-advanced-layer-types/index.html), it's not mentioned why one-hot encoding is not used and why output layer has 10 units but not 1.

The material has 
> To calculate crossentropy loss for data that has its classes represented by integers (i.e., not one-hot encoded), we use the SparseCategoricalCrossentropy() function:

but it's too late to introduce here. 

We should explain why one-hot encoding is not used at the point where building the architecture.

Reference:
1. https://keras.io/api/losses/probabilistic_losses/#sparsecategoricalcrossentropy-class

### PR DESCRIPTION
**Pull request by svenvanderburg**
Fixes #292

### CODE REVIEW

File: episodes/4-advanced-layer-types.Rmd [hunk 1]
```diff
@@ -340,7 +340,14 @@ We compile the model using the adam optimizer (other optimizers could also be us
 Similar to the penguin classification task, we will use the crossentropy function to calculate the model's loss.
 This loss function is appropriate to use when the data has two or more label classes.
 
-To calculate crossentropy loss for data that has its classes represented by integers (i.e., not one-hot encoded), we use the SparseCategoricalCrossentropy() function:
+Remember that our target class is represented by a single integer, whereas the output of our network has 10 nodes, one for each class.
+So, we should have actually one-hot encoded the targets and used a softmax activation for the neurons in our output layer!
+Luckily, there is a quick fix to calculate crossentropy loss for data that
+has its classes represented by integers, the SparseCategoricalCrossentropy().
```
Comment (tobyhodges) [inline episodes/4-advanced-layer-types.Rmd:346]:
```suggestion
has its classes represented by integers, the `SparseCategoricalCrossentropy()` function. 
```

File: episodes/4-advanced-layer-types.Rmd [hunk 2]
```diff
@@ -340,7 +340,14 @@ We compile the model using the adam optimizer (other optimizers could also be us
 Similar to the penguin classification task, we will use the crossentropy function to calculate the model's loss.
 This loss function is appropriate to use when the data has two or more label classes.
 
-To calculate crossentropy loss for data that has its classes represented by integers (i.e., not one-hot encoded), we use the SparseCategoricalCrossentropy() function:
+Remember that our target class is represented by a single integer, whereas the output of our network has 10 nodes, one for each class.
+So, we should have actually one-hot encoded the targets and used a softmax activation for the neurons in our output layer!
+Luckily, there is a quick fix to calculate crossentropy loss for data that
+has its classes represented by integers, the SparseCategoricalCrossentropy(). 
+Adding the argument `from_logits=True` accounts for the fact that the output has a linear activation instead of softmax.
+This is what is often done in practice, because you do not have to worry about one-hot encoding.
```
Comment (tobyhodges) [inline episodes/4-advanced-layer-types.Rmd:348]:
```suggestion
This is what is often done in practice, because it spares you from having to worry about one-hot encoding.
```

**Review by tobyhodges — APPROVED** [review]
Added a few suggestions that I hope will make the explanations clearer.

**Comment by svenvanderburg** [comment]
Thanks @tobyhodges I implemented your suggestions.
